NATIVE_POSTGRES_JSON_COMPARED_TO_EQL.md

```diff
@@ -0,0 +1,510 @@
+# Native Postgres JSON(B) Compared to EQL
+
+EQL supports a subset of functionality supported by the native Postgres JSON(B) functions and operators. The following examples compare native Postres JSON(B) functions and operators to the related functionality in EQL.
+
+## `json ->> text` → `text` and `json -> text` → `jsonb`/`json`
+
+### Native Postgres JSON(B)
+
+```sql
+-- `->` (returns JSON(B))
+SELECT plaintext_jsonb->'field_a' FROM examples;
+
+-- `->>` (returns text)
+SELECT plaintext_jsonb->>'field_a' FROM examples;
+```
+
+### EQL
+
+EQL JSONB functions accept an eJSONPath as an argument (instead of using `->`/`->>`) for lookups.
+
+#### Decryption example
+
+`cs_ste_vec_value_v1` returns the Plaintext EQL payload to the client.
+
+```sql
+SELECT cs_ste_vec_value_v1(encrypted_jsonb, $1) FROM examples;
+```
+
+```javascript
+// Assume that examples.encrypted_jsonb has JSON objects with the shape:
+{
+  "field_a": 100
+}
+
+// `$1` is the EQL plaintext payload for the eJSONPath `$.field_a`:
+{
+  "k": "pt",
+  "p": "$.field_a",
+  "i": {
+    "t": "examples",
+    "c": "encrypted_jsonb"
+  },
+  "v": 1,
+  "q": "ejson_path"
+}
+```
+
+#### Comparison example
+
+`cs_ste_vec_term_v1` returns an ORE term for comparison.
+
+```sql
+SELECT * FROM examples
+WHERE cs_ste_vec_term_v1(examples.encrypted_jsonb, $1) > cs_ste_vec_term_v1($2)
+```
+
+```javascript
+// Assume that examples.encrypted_jsonb has JSON objects with the shape:
+{
+  "field_a": 100
+}
+
+// `$1` is the EQL plaintext payload for the eJSONPath `$.field_a`:
+{
+  "k": "pt",
+  "p": "$.field_a",
+  "i": {
+    "t": "examples",
+    "c": "encrypted_jsonb"
+  },
+  "v": 1,
+  "q": "ejson_path"
+}
+
+// `$2` is the EQL plaintext payload for the ORE term to compare against (in this case, the ORE term for the integer `123`):
+{
+  "k": "pt",
+  "p": "123",
+  "i": {
+    "t": "examples",
+    "c": "encrypted_jsonb"
+  },
+  "v": 1,
+  "q": "ste_vec"
+}
+```
+
+## `json ->> int` → `text` and `json -> int` → `jsonb`/`json`
+
+### Native Postgres JSON(B)
+
+```sql
+-- `->` (returns JSON(B))
+SELECT plaintext_jsonb->0 FROM examples;
+
+-- `->>` (returns text)
+SELECT plaintext_jsonb->>0 FROM examples;
+```
+
+### EQL
+
+EQL JSONB functions accept an eJSONPath as an argument (instead of using `->`/`->>`) for lookups.
```

Comment:
Same here — define or link eJSONPath.

Reply:
Fixed in https://github.com/cipherstash/encrypt-query-language/pull/61/commits/beabf2bd22ddd7489755bcb60e25aaaadc48bc20 and https://github.com/cipherstash/encrypt-query-language/pull/61/commits/9224ad68fcc389a792f253397b66f6287f2205c8.